quickwit/quickwit-query/src/query_ast/term_set_query.rs

```diff
@@ -36,13 +36,20 @@ pub struct TermSetQuery {
     pub terms_per_field: HashMap<String, BTreeSet<String>>,
 }
 
+fn is_term_str(term: &Term) -> bool {
+    let val = term.value();
+    let typ = val.json_path_type().unwrap_or_else(|| val.typ());
+    typ == Type::Str
+}
+
 impl TermSetQuery {
     fn make_term_iterator(
```

Comment:
This function has a very non trivial purpose and a retrun type. It needs a comment.